solutions/USACO feb17-circlecross.mdx

```diff
@@ -2,223 +2,80 @@
 id: usaco-712
 source: USACO Bronze 2017 February
 title: Why Did the Cow Cross the Road II
-author: Benjamin Qi
+author: Benjamin Qi, Danh Ta Chi Thanh
 ---
 
+## Solution 1
+
 [Official Analysis](http://www.usaco.org/current/data/sol_circlecross_bronze_feb17.html)
 
+## Solution 2
+
+Let the first occurrence of a letter be the cow's entry and the last occurrence her exit. Then, we count the number of pairs $(cow_1, cow_2)$ such that $cow_1$ enters before $cow_2$ and leaves somewhere between $cow_2$'s entry and exit. 
```

Comment:
isn't this equivalent to the official analysis?

Reply:
I don't know

Reply:
> isn't this equivalent to the official analysis?

Yeah, it seems that it is similar to the official analysis, so should I remove the ## Solution 2 line?

Reply:
ok

Reply:
Removed.

Reply:
Actually, maybe we can have separate headers for the N^4 and N^2 versions. Like this:

> Official Analysis
> 
> ## $\mathcal{O}(N^4)$ Solution
> 
> (explanation)
> 
> ## $\mathcal{O}(N^2)$ Solution
> 
> Alternatively, we can store the positions of entry and exit points for each of the cows ...

Though the explanation above the N^4 solution seems to correspond to the N^2 solution.

Reply:
> still nothing in explanation of solution 1 that specifically refers to the N^4 code instead of the N^2 code

Reply:
> still nothing in explanation of solution 1 that specifically refers to the N^4 code instead of the N^2 code

Reply:
> still nothing in this explanation that specifically refers to the N^4 code rather than the N^2 code

Reply:
So I have to add some comments about how the N^4 solution can be improved to make it N^2?

Reply:
It's not clear that this explanation corresponds to the N^4 code, which makes no use of `Start` or `End` arrays. @frodakcin what do you think?

Reply:
You're definitely right. Maybe we can have the N^2 solution first, and say smth along the lines of "Alternatively, due to the low constraints, one can solve the problem in N^4 by iterating over all pairs of intersecting paths, and checking whether they represent two cows."

Reply:
> You're definitely right. Maybe we can have the N^2 solution first, and say smth along the lines of "Alternatively, due to the low constraints, one can solve the problem in N^4 by iterating over all pairs of intersecting paths, and checking whether they represent two cows."

Yeah, this seems appropriate since the Bronze division does not require us to strictly follow the O(N^2) solution, so we can use the O(N^4) solution for simplicity and intuition.

Reply:
I'll edit.

Reply:
The solution is detailed, should we add star that problem to recommend the users solve that first? @bqi343

Reply:
idk, I have no idea why some problems are starred and others aren't :eyes:

Reply:
I'm still wondering when my pull request will be merged 👀